fe/fe-core/src/main/java/org/apache/doris/analysis/SelectStmt.java

```diff
@@ -613,6 +617,102 @@ private void whereClauseRewrite() {
+        // left outer join -> inner join
+        // next table:
+        // full outer join -> left outer join
+        // right outer join -> inner join
+        UnaryOperator<JoinOperator> convertSelf = joinOp -> {
+            if (joinOp.isFullOuterJoin()) {
+                return JoinOperator.RIGHT_OUTER_JOIN;
+            } else if (joinOp.isLeftOuterJoin()) {
+                return JoinOperator.INNER_JOIN;
+            }
+            return joinOp;
+        };
+        UnaryOperator<JoinOperator> convertNext = joinOp -> {
+            if (joinOp.isFullOuterJoin()) {
+                return JoinOperator.LEFT_OUTER_JOIN;
+            } else if (joinOp.isRightOuterJoin()) {
+                return JoinOperator.INNER_JOIN;
+            }
+            return joinOp;
+        };
+
+        for (int i = 0; i < fromClause.size(); i++) {
+            if (fromClause == null)
+                return;
+
+            TableRef tableRef = fromClause.get(i);
+            if (tableRef == null || tableRef.getName() == null)
+                continue;
+            if (notNullTables.contains(tableRef.getName().getTbl())) {
+                tableRef.setJoinOp(convertSelf.apply(tableRef.getJoinOp()));
+
+                if (i < fromClause.size() - 1) {
+                    TableRef nextTableRef = fromClause.get(i + 1);
+                    nextTableRef.setJoinOp(convertNext.apply(nextTableRef.getJoinOp()));
+                }
+            }
+        }
+
+        // Clean outerJoinedTupleIds/fullOuterJoinedTupleIds/fullOuterJoinedConjuncts in
+        // globalState.
+        // Make predicate can be pushed down after conversion.
+        ArrayList<ExprId> conjunctIds = Lists.newArrayList();
+        analyzer.getFullOuterJoinedConjuncts().forEach((id, tableRef) -> {
+            if (tableRef != null && tableRef.joinOp.isRightOuterJoin() || tableRef.joinOp.isInnerJoin()
+                    || (tableRef.leftTblRef != null
+                            && (tableRef.leftTblRef.joinOp.isLeftOuterJoin()
+                                    || tableRef.joinOp.isInnerJoin())))
+                conjunctIds.add(id);
+        });
+        conjunctIds.forEach(analyzer.getFullOuterJoinedConjuncts()::remove);
+
+        Consumer<Map<TupleId, TableRef>> remove = (tupleIds) -> {
+            ArrayList<TupleId> ids = Lists.newArrayList();
+            tupleIds.forEach((id, tableRef) -> {
+                if (tableRef != null && tableRef.joinOp.isRightOuterJoin() || tableRef.joinOp.isInnerJoin()
+                        || (tableRef.leftTblRef != null
+                                && (tableRef.leftTblRef.joinOp.isLeftOuterJoin()
+                                        || tableRef.joinOp.isInnerJoin())))
+                    ids.add(id);
+            });
+            ids.forEach(tupleIds::remove);
+        };
+        remove.accept(analyzer.getFullOuterJoinedTupleIds());
+        remove.accept(analyzer.getOuterJoinedTupleIds());
```

Comment:
1. Do we need to remove conjuncts from `org.apache.doris.analysis.Analyzer.GlobalState#fullOuterJoinedConjuncts`?
2. Do we need to remove TupleId from `org.apache.doris.analysis.Analyzer.GlobalState#outerJoinedMaterializedTupleIds`?

@EmmyMiao87 could you help to check out whether it is correct?